src/error.rs

```diff
@@ -1,4 +1,7 @@
 //! Common error type definitions
+
+use core::fmt;
+use thiserror_no_std::Error;
```

Comment:
This `use` statement will not work when the `thiserror-no-std` feature is disabled; compilation will fail.

See my next review comment for a suggested fix